### ISSUE TITLE
`AttributeError` in a Django project

### ISSUE THREAD

**Issue by ShreehariVaasishta** [thebjorn/pydeps#139]
I'm running the pydeps inside the project app `project/project/__init__.py` and getting the following error **` AttributeError: 'NoneType' object has no attribute 'is_package'`**.
Complete traceback can be found in this text file- [error-traceback.txt](https://github.com/thebjorn/pydeps/files/8501004/traceback.txt)
**Note:** I'm using **`Python 3.8.10`**

**Comment by thebjorn** [comment]
the traceback..
```
$ pydeps .
Traceback (most recent call last):
  File "/mnt/d/venv12/bin/pydeps", line 8, in <module>
    sys.exit(pydeps())
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/pydeps.py", line 150, in pydeps
    return _pydeps(inp, **_args)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/pydeps.py", line 32, in _pydeps
    dep_graph = py2depgraph.py2dep(trgt, **kw)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 214, in py2dep
    mf.run_script(dummy.fname)
  File "/usr/lib/python3.8/modulefinder.py", line 163, in run_script
    self.load_module('__main__', fp, pathname, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 75, in load_module
    self.scan_code(co, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 103, in scan_code
    self._safe_import_hook(name, m, fromlist, level=level)
  File "/usr/lib/python3.8/modulefinder.py", line 392, in _safe_import_hook
    self.import_hook(name, caller, [sub], level=level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 27, in import_hook
    self.ensure_fromlist(m, fromlist)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 171, in ensure_fromlist
    submod = self.import_module(sub, subname, module)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 138, in import_module
    module = mf27.ModuleFinder.import_module(self, partnam, fqname, parent)
  File "/usr/lib/python3.8/modulefinder.py", line 325, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 75, in load_module
    self.scan_code(co, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 103, in scan_code
    self._safe_import_hook(name, m, fromlist, level=level)
  File "/usr/lib/python3.8/modulefinder.py", line 377, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 23, in import_hook
    m = self.load_tail(q, tail)
  File "/usr/lib/python3.8/modulefinder.py", line 253, in load_tail
    m = self.import_module(head, mname, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 138, in import_module
    module = mf27.ModuleFinder.import_module(self, partnam, fqname, parent)
  File "/usr/lib/python3.8/modulefinder.py", line 325, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 41, in load_module
    module = self.load_package(fqname, pathname)
  File "/usr/lib/python3.8/modulefinder.py", line 481, in load_package
    self.load_module(fqname, fp, buf, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 75, in load_module
    self.scan_code(co, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 103, in scan_code
    self._safe_import_hook(name, m, fromlist, level=level)
  File "/usr/lib/python3.8/modulefinder.py", line 377, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 23, in import_hook
    m = self.load_tail(q, tail)
  File "/usr/lib/python3.8/modulefinder.py", line 253, in load_tail
    m = self.import_module(head, mname, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 138, in import_module
    module = mf27.ModuleFinder.import_module(self, partnam, fqname, parent)
  File "/usr/lib/python3.8/modulefinder.py", line 325, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 75, in load_module
    self.scan_code(co, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 103, in scan_code
    self._safe_import_hook(name, m, fromlist, level=level)
  File "/usr/lib/python3.8/modulefinder.py", line 377, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 23, in import_hook
    m = self.load_tail(q, tail)
  File "/usr/lib/python3.8/modulefinder.py", line 253, in load_tail
    m = self.import_module(head, mname, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 138, in import_module
    module = mf27.ModuleFinder.import_module(self, partnam, fqname, parent)
  File "/usr/lib/python3.8/modulefinder.py", line 325, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 75, in load_module
    self.scan_code(co, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 103, in scan_code
    self._safe_import_hook(name, m, fromlist, level=level)
  File "/usr/lib/python3.8/modulefinder.py", line 377, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 23, in import_hook
    m = self.load_tail(q, tail)
  File "/usr/lib/python3.8/modulefinder.py", line 253, in load_tail
    m = self.import_module(head, mname, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 138, in import_module
    module = mf27.ModuleFinder.import_module(self, partnam, fqname, parent)
  File "/usr/lib/python3.8/modulefinder.py", line 325, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 41, in load_module
    module = self.load_package(fqname, pathname)
  File "/usr/lib/python3.8/modulefinder.py", line 481, in load_package
    self.load_module(fqname, fp, buf, stuff)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 145, in load_module
    module = mf27.ModuleFinder.load_module(
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 75, in load_module
    self.scan_code(co, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 103, in scan_code
    self._safe_import_hook(name, m, fromlist, level=level)
  File "/usr/lib/python3.8/modulefinder.py", line 377, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/mf27.py", line 23, in import_hook
    m = self.load_tail(q, tail)
  File "/usr/lib/python3.8/modulefinder.py", line 253, in load_tail
    m = self.import_module(head, mname, m)
  File "/mnt/d/venv12/lib/python3.8/site-packages/pydeps/py2depgraph.py", line 138, in import_module
    module = mf27.ModuleFinder.import_module(self, partnam, fqname, parent)
  File "/usr/lib/python3.8/modulefinder.py", line 318, in import_module
    fp, pathname, stuff = self.find_module(partname,
  File "/usr/lib/python3.8/modulefinder.py", line 510, in find_module
    return _find_module(name, path)
  File "/usr/lib/python3.8/modulefinder.py", line 79, in _find_module
    if spec.loader.is_package(name):
AttributeError: 'NoneType' object has no attribute 'is_package'
```

**Comment by thebjorn** [comment]
Which version of pydeps are you using (`pydeps --version`)? Is your project installed in your virtualenv (e.g. `pip install -e .`)?  I can't reproduce your problem with the latest pydeps on Python 3.8, so a MCVE would be helpful in tracking down the problem.

**Comment by ShreehariVaasishta** [comment]
Thanks for the response.
```
$ pydeps --version
pydeps v1.10.14
```

If i create a new Django project and try it there, it seems to be working fine. In my existing Django project(closed source) I'm facing the issue. I will check how i can replicate and share the MCVE here in some time.

**Comment by yangkyeongmo** [comment]
I had same issue on pydeps v1.10.14, but error is gone after downgraded to v1.10.0. Currently using python 3.10.4.

**Comment by ShreehariVaasishta** [comment]
> I had same issue on pydeps v1.10.14, but error is gone after downgraded to v1.10.0. Currently using python 3.10.4.

Thank you, downgrading the version has fixed the issue for me.

**Comment by thebjorn** [comment]
A testcase that would demonstrate the problem would be great. Without it I'm pretty much blind when trying to figure out what is going on...

Could one of you try the latest version (v.1.10.16) which has an extra exception handler that "might" work..?

**Comment by thebjorn** [comment]
See #141

**Comment by thebjorn** [comment]
This should be fixed in the latest version (v1.10.17) available on PyPI.
